DIPs/DIP1xxx-mvf.md

```diff
@@ -0,0 +1,605 @@
+
+    // Getter `@property` function that returns an lvalue.
+    // No setter.
+    //--------------------------------------------------------------
+    ref int prop1() @property
+    {
+        return field;
+    }
+
+    // Getter `@property` function that returns an lvalue.
+    // Accompanying setter `@property` function.
+    //--------------------------------------------------------------
+    ref int prop2() @property
+    {
+        return field;
+    }
+
+    ref int prop2(int value) @property
+    {
+        return field = value;
+    }
+
+    // Ordinary function that returns an lvalue.
+    // No accompanying setter
+    //--------------------------------------------------------------
+    ref int prop3()             // NOTE: no `@property` attribute
+    {
+        return field;
+    }
+}
+
+void main()
+{
+    S s;
+
+    s.prop1 += 1;               // Existing Implementation:
+                                // OK: Operator is applied to the return value.
+                                //-------------------------------------------------------------------
+                                // OK: Special Case 2: Due to the fact the getter returns an lvalue,
+                                // the expression is not lowered, resulting in the existing
+                                // implementation's behavior, maintaining the status quo.
+
+    s.prop2 += 1;               // Existing Implementation:
+                                // OK: Operator is applied to the return value.  Setter is
+                                // never called.
+                                //-------------------------------------------------------------------
+                                // Proposed Implementation:
+                                // OK: Special Case 2: Due to the fact the getter returns an lvalue,
+                                // the expression is not lowered, resulting in the existing
+                                // implementation's behavior, maintaining the status quo.
+
+    s.prop3 += 1;               // Existing Implementation:
+                                // OK: Due to existing optional parentheses feature, operator
+                                // is applied to the return value.
+                                //-------------------------------------------------------------------
+                                // Proposed Implementation:
+                                // OK: Special Cases 1 and 2: Due to the absence of the `@property`
+                                // attribute and the fact that the getter returns an lvalue, the
+                                // expression is not lowered, resulting in the existing
+                                // implementation's behavior, maintaining the status quo.
+}
+```
+
+Despite the unfortunate inconsistency of this proposal treating rvalue properties and lvalue properties differently, it is necessary to prevent breakage and maintain the status quo.
```

Comment:
This is unfortunate indeed. Is a deprecation path viable here maybe? So if there's code like this:
```
struct S {
  int value;
  @property ref int prop() { return value; }
  @property ref int prop(int v) { return value = v; }
}

S s;
s.prop += 3;
```

Then something like: `Deprecated. op= on a ref returning property. Please remove the @property attribute from the getter.`

But that would require that an @property name can have a non-@property name that is an overload, which I believe you cannot do. But maybe good enough reason here to allow this case until the deprecation period is over?

Reply:
I don't think that pattern appears very often in code because it's only necessary to have the `ref` getter to accomplish the task.  What I hope will happen is that after this DIP is approved and implemented, users will refactor their code moving away from `ref` properties.  Then the pattern will become even less likely, and then we can issue a deprecation or change in behavior with less disruption.